src/gen_hook.erl

```diff
@@ -210,31 +210,24 @@ run_hook([Handler | Ls], Acc, Params, Key) ->
             run_hook(Ls, NewAcc, Params, Key);
         {stop, NewAcc} ->
             {stop, NewAcc};
-        Other ->
-            ?MODULE:error_running_hook(Other, Handler, Acc, Params, Key),
+        {error, Info} ->
+            ?MODULE:error_running_hook(Info, Handler, Acc, Params, Key),
             run_hook(Ls, Acc, Params, Key)
     end.
 
 -spec apply_hook_function(#hook_handler{}, hook_acc(), hook_params()) ->
-    hook_fn_ret() | {'EXIT', Reason :: any()}.
+    safely:catch_result(hook_fn_ret()).
```

Comment:
This type is renamed to `result`.

Reply:
Ok I've changed the name and the definition anyway, maybe a tuple tagged by `error` is way too generic all over erlang code. But it's so sad nor xref nor dialyzer can catch such an obvious one 🤦🏽

Reply:
Dialyzer can do it with `-Wunknown`, but this would result in a lot of warnings. We could try to do it soon - actually this was one of the reasons to get rid of the dynamically compiled modules.

Reply:
Right... gonna try that out when I find a free minute, thanks for the tip 🤔